### PR TITLE
Clean up debugging options

### PR DESCRIPTION
**Pull request by nnethercote**
I found various sub-optimal things when I was looking at option handling.

### CODE REVIEW

File: src/librustc_session/options.rs [hunk 1]
```diff
@@ -814,12 +811,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
         "for every macro invocation, print its name and arguments"),
     debug_macros: bool = (false, parse_bool, [TRACKED],
         "emit line numbers debug info inside macros"),
-    generate_arange_section: bool = (true, parse_bool, [TRACKED],
-        "generate DWARF address ranges for faster lookups"),
```
Comment (michaelwoerister) [inline src/librustc_session/options.rs]: Why is this removed?

Comment (michaelwoerister) [reply src/librustc_session/options.rs]: Oh, I read the comment on `parse_bool` now. It seems like the behavior of `parse_bool` is not intentional. I'm OK with removing the kind of outdated `-Z incremental-queries` flags, but for `-Z generate-arange-section` it seems useful to enable opting out.

Comment (nnethercote) [reply src/librustc_session/options.rs]: So that will require either changing it to `-Z no-generate-arange-section`. Is that ok?

Comment (nnethercote) [reply src/librustc_session/options.rs]: I have made this change, and split it into a separate commit.

**Comment by nnethercote** [comment]
The `-Z` changes seem like they should be mentioned in the release notes for whatever version this landed in. How do I request that?

**Comment by lzutao** [comment]
cc @Centril ^

**Comment by Centril** [comment]
`-Z` changes should only affect nightly and we don't include that in relnotes.
